### PR TITLE
WSO2 App Server Utils tests and refactored project.

### PR DESCRIPTION
**Pull request by chirangaalwis**

### CODE REVIEW

**Comment by chirangaalwis** [comment]
Associated JIRAs

https://wso2.org/jira/browse/WSAS-2208
https://wso2.org/jira/browse/WSAS-2204

File: integration-tests/pom.xml [hunk 1]
```diff
@@ -43,97 +39,145 @@
...
+            <scope>test</scope>
         </dependency>
+
+        <!-- Test dependencies -->
         <dependency>
-            <groupId>org.apache.thrift</groupId>
-            <artifactId>libthrift</artifactId>
-            <version>0.9.3</version>
+            <groupId>org.apache.tomcat</groupId>
+            <artifactId>tomcat-catalina</artifactId>
```
Comment (callkalpa) [inline integration-tests/pom.xml]: Why do we need a tomcat dependency for integration tests?

File: integration-tests/src/test/java/org/wso2/appserver/test/integration/utils/ConfigurationLoaderTestCase.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This Java class defines integration test cases for WSO2 Application Server specific configuration loading.
+ *
+ * @since 6.0.0
+ */
+public class ConfigurationLoaderTestCase extends TestBase {
+    private static final String sampleContextRoot = "/configuration-loader-" + System.getProperty("appserver.version");
```
Comment (callkalpa) [inline integration-tests/src/test/java/org/wso2/appserver/test/integration/utils/ConfigurationLoaderTestCase.java]: How do we deploy the configuration-loader webapp to app server?

**Comment by chirangaalwis** [comment]
We don't need it now, it was used earlier for accessing Tomcat listener methods in integration tests. Removed in last commit.

File: integration-tests/src/test/java/org/wso2/appserver/test/integration/utils/ConfigurationLoaderTestCase.java (hunk 2 above)
Comment (chirangaalwis) [inline integration-tests/src/test/java/org/wso2/appserver/test/integration/utils/ConfigurationLoaderTestCase.java]: This is performed by maven-dependency-plugin at pre-integration-test phase.

```
<execution>
 +                        <id>add-test-webapps</id>
 +                        <phase>pre-integration-test</phase>
 +                        <goals>
 +                            <goal>copy-dependencies</goal>
 +                        </goals>
 +                        <configuration>
 +                            <includeGroupIds>org.wso2.appserver</includeGroupIds>
 +                            <includeArtifactIds>configuration-loader</includeArtifactIds>
 +                            <outputDirectory>${project.build.directory}/${distribution.file.name}/webapps
 +                            </outputDirectory>
 +                        </configuration>
 +                    </execution>
```

File: modules/appserver-utils/src/test/java/org/wso2/appserver/AppServerWebAppConfigurationTest.java [hunk 3]
```diff
@@ -34,26 +45,38 @@
...
+
+        Optional<AppServerWebAppConfiguration> effective = ContextConfigurationLoader.getContextConfiguration(context);
+        if (effective.isPresent()) {
+            Assert.assertTrue(compare(effective.get(), prepareDefault()));
+        } else {
+            Assert.fail();
+        }
     }
 
     private static AppServerWebAppConfiguration prepareDefault() {
```
Comment (callkalpa) [inline modules/appserver-utils/src/test/java/org/wso2/appserver/AppServerWebAppConfigurationTest.java]: please make the method name complaint with the naming convention

File: modules/appserver-utils/src/test/java/org/wso2/appserver/AppServerWebAppConfigurationTest.java [hunk 4]
```diff
@@ -109,35 +132,43 @@ private static SSOConfiguration prepareSSOConfiguration() {
         return ssoConfiguration;
     }
 
+    private static StatsPublisherConfiguration prepareStatsPublisherConfiguration() {
```
Comment (callkalpa) [inline modules/appserver-utils/src/test/java/org/wso2/appserver/AppServerWebAppConfigurationTest.java]: please make the method name complaint with the naming convention

**Pull request merged**